src/main/java/com/google/error_analyzer/backend/Database.java

```diff
@@ -0,0 +1,160 @@
+    private static final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+    private static final int windowSize = 10;
+    private static final Logger logger = LogManager.getLogger(Database.class);
+
+    //search db using keywords and return searchHits having highlight field added 
+    public ArrayList < SearchHit > fullTextSearch(String fileName, String searchString, String field) throws IOException {
+        int offset = 0;
+        SearchHit[] searchHits = null;
+        ArrayList < SearchHit > searchResult = new ArrayList();
+        while (searchHits == null || searchHits.length != 0) {
+            SearchRequest searchRequest = new SearchRequest(fileName);
+            SimpleQueryStringBuilder simpleQueryBuilder = QueryBuilders.simpleQueryStringQuery(searchString);
+            searchSourceBuilder.query(simpleQueryBuilder).size(windowSize).from(offset);
+            HighlightBuilder highlightBuilder = addHighLighter(field);
+            searchSourceBuilder.highlighter(highlightBuilder);
+            searchRequest.source(searchSourceBuilder);
+            SearchResponse searchResponse = client.search(searchRequest, RequestOptions.DEFAULT);
+            SearchHits hits = searchResponse.getHits();
+            searchHits = hits.getHits();
+            for (SearchHit hit: searchHits) {
+                searchResult.add(hit);
+            }
+            offset += windowSize;
+        }
+        return searchResult;
+    }
+    // highlight searched text
+    private HighlightBuilder addHighLighter(String field) {
+        HighlightBuilder highlightBuilder = new HighlightBuilder().preTags("<b>").postTags("</b>");
+        HighlightBuilder.Field highlightTitle = new HighlightBuilder.Field(field);
+        highlightTitle.highlighterType("unified");
+        highlightBuilder.field(highlightTitle);
+        return highlightBuilder;
+    }
+
+    // return ArrayList of hit ids corresponding to given searchhit list
+    public ArrayList < String > hitId(SearchHit[] searchHits) throws IOException {
+        ArrayList < String > ids = new ArrayList();
+        for (SearchHit hit: searchHits) {
+            String id = hit.getId();
+            ids.add(id);
+        }
+        return ids;
+    }
+
+    // return ArrayList of content for specified field  corresponding to given searchhit list
+    public ArrayList < String > hitFieldContent(SearchHit[] searchHits, String field) throws IOException {
+        ArrayList < String > fieldContent = new ArrayList();
+        for (SearchHit hit: searchHits) {
+            String resultString = String.valueOf(hit.getSourceAsMap().get(field));
+            fieldContent.add(resultString);
+        }
+        return fieldContent;
+    }
+
+    //search db using user provided regex and return searchHits having highlight field added
+    public SearchHit[] regexQuery(String filename, String regex) {
+        return new SearchHit[0];
```

Comment:
not implemented yet?

Reply:
Rythm have implemented it .will be there in her PR